### PR TITLE
Use Composition API to add/remove from composition

### LINKED ISSUE #3117: [DisplayLayout] Object removal from object tree doesn't consistently remove object from Display Layout in main window
Overview to reproduce error:
1. Add an item to display layout, then "Save and Finish Editing"
2. Select the child item in the object tree so that it's active in the main window
3. Right-click and remove object. Notice it was removed from object tree but not display layout.

Detailed steps to reproduce error:
1. Add a display layout to "My Items" folder
2. In Display Layout main window, click "Save and Finish Editing"
3. Add a "State Generator" object to the Display Layout in the object tree
4. Select the Dispay Layout in the object tree so it's active in the main window
5. Note that you can see the State Generator in the display layout in the upper left corner of the main window
6. Select the State Generator in the object tree so it's active in the main window
7. Right-click the State Generator and select Remove from the drop-down menu
8. Note that the State Generator is removed from the object tree, which is expected
9. Select the Display Layout in the object tree so it's active in the main window
10. Note that the State Generator was not removed from the display layout in the main window, which is unexpected behavior.

Expected behavior: When the state generator is removed from the display layout in the object tree, it should also be removed from the display layout in the main window.

Actual behavior: When the state generator is removed from the display layout in the object tree, it's not removed from the display layout in the main window.

(Ubuntu 16.04 - Firefox 77.0.1 64-bit, Chrome Version 70.0.3538.102 64-bit)

### LINKED ISSUE #5860: Refactor code to use composition API
#### Summary
There are [several](https://github.com/nasa/openmct/blob/cb8e09c9f9ec176884c8b8d193a9f8f01667077e/src/plugins/remove/RemoveAction.js#L89) [instances](https://github.com/nasa/openmct/blob/cb8e09c9f9ec176884c8b8d193a9f8f01667077e/src/plugins/charts/scatter/ScatterPlotView.vue#L119) [in](https://github.com/nasa/openmct/blob/cb8e09c9f9ec176884c8b8d193a9f8f01667077e/src/plugins/gauge/components/Gauge.vue#L605) [our](https://github.com/nasa/openmct/blob/cb8e09c9f9ec176884c8b8d193a9f8f01667077e/src/plugins/displayLayout/components/DisplayLayout.vue#L515) codebase where domain object composition is being changed by directly mutating the `composition` property of domain objects. This is problematic for two reasons:
1. It will not trigger composition listeners.
2. It will not work at all with anything other than the default composition provider.

We see a lot of bugs that are some version of "removing an object from X does not update Y". This issue may underlie a lot of these kinds of issues that keep coming up. There was legacy code that used to plaster over this issue by automatically detecting composition changes in object mutation in order to bridge the legacy and current APIs, but this code has been removed along with the legacy codebase.

#### Expected vs Current Behavior
Object composition operations should _always_ use the composition API.

### PR DESCRIPTION
**Pull request by ozyx**
Closes #5860 #3482 #3117  

### Describe your changes:

- Refactors some areas of our code to use the composition API instead of directly mutating object composition.
- Fixes a small issue where RemoveAction would re-use old transactions under certain conditions.
- Fixes some logic regarding PlotSeries removal for ScatterPlots
- Fixes #3117 which regressed after we implemented transactions for RemoveAction in #5827
- adds e2e tests for scatterPlot, gauge, displayLayout and flexibleLayout

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [x] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [x] Changes address original issue?
* [x] Tests included and/or updated with changes?
* [x] Command line build passes?
* [x] Has this been smoke tested?
* [x] Testing instructions included in associated issue OR is this a dependency/testcase change?

### Reviewer Checklist

* [x] Changes appear to address issue?
* [x] Reviewer has tested changes by following the provided instructions?
* [x] Changes appear not to be breaking changes?
* [x] Appropriate automated tests included?
* [x] Code style and in-line documentation are appropriate?
* [x] Has associated issue been labelled unverified? (only applicable if this PR closes the issue)
* [x] Has associated issue been labelled bug? (only applicable if this PR is for a bug fix)

### CODE REVIEW

File: src/plugins/flexibleLayout/components/flexibleLayout.vue [hunk 1]
```diff
@@ -366,6 +383,24 @@ export default {
...
+                this.transaction = this.openmct.objects.startTransaction();
+            }
+        },
+        endTransaction() {
+            if (!this.transaction) {
+                return;
+            }
+
+            return this.transaction.commit()
+                .catch(error => {
```
Comment (scottbell) [inline src/plugins/flexibleLayout/components/flexibleLayout.vue:398]: same as above 👆

**Review by scottbell — APPROVED** [review]
LGTM except for the small stuff below

**Comment by scottbell** [comment]
I tested #3117 and it worked well.

**Review by scottbell — APPROVED** [review]
Had some small stuff, but overall LGTM

**Comment by ozyx** [comment]
Going to refactor as per @scottbell 's suggestions

**Review by jvigliotta — APPROVED** [review]
Some comments, but nothing that major requires any changes. We can get this in.

File: e2e/tests/functional/plugins/gauge/gauge.e2e.spec.js [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    test.beforeEach(async ({ page }) => {
+        // Open a browser, navigate to the main page, and wait until all networkevents to resolve
+        await page.goto('./', { waitUntil: 'networkidle' });
+
+        // Create the gauge
+        gauge = await createDomainObjectWithDefaults(page, { type: 'Gauge' });
+    });
+
+    test('Can add and remove telemetry sources @unstable', async ({ page }) => {
+        const editButtonLocator = page.locator('button[title="Edit"]');
```
[the remark this answers is not in the document]
Comment (ozyx) [reply e2e/tests/functional/plugins/gauge/gauge.e2e.spec.js:43]: actually yeah, that's a good idea here. let me update these

File: e2e/tests/functional/plugins/flexibleLayout/flexibleLayout.e2e.spec.js [hunk 3]
```diff
@@ -54,13 +55,81 @@ test.describe('Testing Flexible Layout @unstable', () => {
...
+
+        // Expand the Flexible Layout so we can remove the sine wave generator
+        await page.locator('.c-tree__item.is-navigated-object .c-disclosure-triangle').click();
+
+        // Bring up context menu and remove
+        await page.locator('.c-tree__item.is-alias .c-tree__item__name:text("Test Sine Wave Generator")').first().click({ button: 'right' });
+        await page.locator('li[role="menuitem"]:has-text("Remove")').click();
+        await page.locator('button:has-text("OK")').click();
+
+        // delete
```
[the remark this answers is not in the document]
Comment (ozyx) [reply e2e/tests/functional/plugins/flexibleLayout/flexibleLayout.e2e.spec.js:93]: i'll fix it!

**Pull request merged**